deploy/aws/java11Exec/src/main/java/poseidon/App.java

```diff
@@ -0,0 +1,136 @@
+package poseidon;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Base64;
+import java.util.Map;
+
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.apigatewaymanagementapi.AmazonApiGatewayManagementApi;
+import com.amazonaws.services.apigatewaymanagementapi.AmazonApiGatewayManagementApiClientBuilder;
+import com.amazonaws.services.apigatewaymanagementapi.model.PostToConnectionRequest;
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayV2WebSocketEvent;
+import com.google.gson.Gson;
+
+// AwsFunctionRequest contains the java files that needs to be executed.
+class AwsFunctionRequest {
+    String[] cmd;
+    Map<String, String> files;
+}
+
+// WebSocketMessageType are the types of messages that are being send back over the WebSocket connection.
+enum WebSocketMessageType {
+    WebSocketOutputStdout("stdout"),
+    WebSocketOutputStderr("stderr"),
+    WebSocketOutputError("error"),
+    WebSocketExit("exit");
+
+    private final String typeName;
+
+    WebSocketMessageType(String name) {
+        this.typeName = name;
+    }
+
+    public String toString() {
+        return typeName;
+    }
+}
+
+// WebSocketMessage is the object that is beeing send over the WebSocket connection.
+class WebSocketMessage {
+    WebSocketMessageType type;
+    String data;
+    int exitCode;
+}
+
+/**
+ * Handler for requests to Lambda function.
+ * This Lambda function executes the passed command with the provided files in an isolated Java environment.
+ */
+public class App implements RequestHandler<APIGatewayV2WebSocketEvent, APIGatewayProxyResponseEvent> {
+
+    // gson helps parse the json objects.
+    private static final Gson gson = new Gson();
+
+    // gwClient is used to send messages back via the WebSocket connection.
+    private AmazonApiGatewayManagementApi gwClient;
+
+    // connectionID helps to identify the WebSocket connection that has called this function.
+    private String connectionID;
+
+    public static final String disableOutputHeaderKey = "disableOutput";
+
+    // disableOutput: If set to true, no output will be sent over the WebSocket connection.
+    private boolean disableOutput = false;
+
+    public APIGatewayProxyResponseEvent handleRequest(final APIGatewayV2WebSocketEvent input, final Context context) {
+        APIGatewayV2WebSocketEvent.RequestContext ctx = input.getRequestContext();
+        this.gwClient = AmazonApiGatewayManagementApiClientBuilder.standard()
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration("https://" + ctx.getDomainName() + "/" + ctx.getStage(), Regions.EU_CENTRAL_1.getName()))
```

Comment:
We might want to add a comment somewhere (in the documentation) that EU Central is hard-coded here.

Reply:
I am fine with both. If you have a quick and easy solution to parse the region from the URL or any other parameter, I would do that. If you spend more than 15 / 30 minutes on that, I would add it to the ReadMe.